src/renderer/apis/commands.ts

```diff
@@ -1,31 +1,119 @@
-import type { RepluggedCommand } from "../../types";
+import type {
+  RepluggedCommand,
+  RepluggedCommandResult,
+  RepluggedCommandSection,
+} from "../../types";
 import { Logger } from "../modules/logger";
+import { channels, guilds, messages } from "../modules/common";
+import { CommandOptionReturn } from "../../types/discord";
+const logger = Logger.api("Commands");
+interface CommandsAndSection {
+  section: RepluggedCommandSection;
+  commands: Map<string, RepluggedCommand>;
+}
 
-const commandsLogger = Logger.api("Commands");
+export const commandAndSections = new Map<string, CommandsAndSection>();
 
-class CommandsAPI extends EventTarget {
-  public commands = new Map<string, RepluggedCommand>();
+export const defaultSection: RepluggedCommandSection = Object.freeze({
+  id: "replugged",
+  name: "Replugged",
+  type: 1,
+  icon: "https://cdn.discordapp.com/attachments/1000955992068079716/1004196106055454820/Replugged-Logo.png",
+});
 
-  public registerCommand(command: RepluggedCommand): void {
-    if (this.commands.has(command.name)) {
-      commandsLogger.error(`Command “${command.name}” is already registered!`);
-      return;
-    }
+async function executeCommand(
+  cmdExecutor:
+    | ((args: CommandOptionReturn[]) => Promise<RepluggedCommandResult> | RepluggedCommandResult)
+    | undefined,
+  args: CommandOptionReturn[],
+): Promise<void> {
+  try {
+    const result = await cmdExecutor?.(args);
+    const currentGuildId = guilds.getGuildId();
+    const currentChannelId = channels.getChannelId(currentGuildId!)!;
 
-    this.commands.set(command.name, command);
-    this.dispatchEvent(new CustomEvent("rpCommandAdded", { detail: { command } }));
-  }
+    if ((!result?.result && !result?.embeds) || !currentChannelId) return;
 
-  public unregisterCommand(name: string): void {
-    if (this.commands.has(name)) {
-      this.dispatchEvent(
-        new CustomEvent("rpCommandRemoved", { detail: { command: this.commands.get(name) } }),
-      );
-      this.commands.delete(name);
+    if (result.send) {
+      void messages.sendMessage(currentChannelId, {
+        content: result.result!,
+        invalidEmojis: [],
+        validNonShortcutEmojis: [],
+        tts: false,
+      });
     } else {
-      commandsLogger.error(`Command “${name}” is not registered!`);
+      const botMessage = messages.createBotMessage({
+        channelId: currentChannelId,
+        content: result.result || "",
+        embeds: result.embeds || [],
+        loggingName: "Replugged",
+      });
+
+      Object.assign(botMessage.author, {
+        username: "Replugged",
+        avatar: "replugged",
+      });
+
+      messages.receiveMessage(currentChannelId, botMessage);
     }
+  } catch (error) {
+    logger.error(error);
+    const currentChannelId = channels.getLastSelectedChannelId()!;
+    const botMessage = messages.createBotMessage?.({
```

Comment:
Can this be sent as a response to the command like with a regular slash command? Ideally it would show the same loading message too.